plugins/ipstack/icon_ipstack/connection/connection.py

```diff
@@ -13,4 +16,18 @@ def connect(self, params):
         self.token = params.get(Input.CRED_TOKEN).get("secretKey")
 
     def test(self):
-        return {"success": True}
+        # Sending a request using the secretKey to test if it is valid
+        url = "http://api.ipstack.com/" + "rapid7.com" + "?access_key=" + self.token + "&output=json"
+
+        invalid_status_codes = ["404", "101", "102", "104", "106"]
```

Comment:
Might be worth adding a comment here to say that these 1xx are custom status code returned from ipstack as in the future people might question why we are checking against these

Reply:
Good idea, will add now!